### PR TITLE
feat: move to scikit-build-core

### LINKED ISSUE #104: Migrate from scikit-build to scikit-build-core
Try migrating from scikit-build to scikit-build-core, which could remove the need for setup.py and MANIFEST.in files.

https://github.com/scikit-build/scikit-build-core/issues/274#issuecomment-1508493949 mentions an additional step that is needed for using setuptools_scm for versioning.

One unknown issue (that works with setuptools specifies data/share/swig subfolder in setup.py but not in pyproject.toml) is whether or not those swig support files that get installed will be included in the correct subfolder when building using scikit-build-core.

### PR DESCRIPTION
**Pull request by henryiii**
Close #104.

I didn't include these files in the [existing wheel](https://inspector.pypi.io/project/swig/4.2.1/packages/17/e7/1834b83e9ef378a753109b716dfc8fdc6da2e42945ce86792b2ccebd9a80/swig-4.2.1-py2.py3-none-macosx_10_9_universal2.whl/):

```
./swig-4.2.1.data/data/LICENSE-SWIG
./swig-4.2.1.data/data/CMakeLists.txt
./swig-4.2.1.data/data/LICENSE-UNIVERSITIES
./swig-4.2.1.data/data/README.md
./swig-4.2.1.data/data/LICENSE-GPL
```

as the licenses are already in the correct place in the diet-info folder, people don't need the CMakeLists.txt or README.md, and these get installed to the _root_ of the environment! For a virtualenv, that's bad:

```console
$ uv venv
$ uv pip install swig
$ ls .venv
CACHEDIR.TAG          LICENSE-GPL           LICENSE-UNIVERSITIES  bin/                  pyvenv.cfg
CMakeLists.txt        LICENSE-SWIG          README.md             lib
```

And for a global install, that's horrible:

```console
$ docker run --rm -it python:3.12 bash
# pip install swig
# ls /usr/local
CMakeLists.txt	LICENSE-GPL  LICENSE-SWIG  LICENSE-UNIVERSITIES  README.md  bin  etc  games  include  lib  man	sbin  share  src
```

The second commit uses a single macOS universal wheel with ARM and Intel tags also included. I could remove that if you'd prefer multiple smaller wheels. As it is, pip will never download the universal wheel, since it will get the most specific one.

### CODE REVIEW

**Comment by nightlark** [comment]
Thank you!! 😃

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
             artifact_suffix: "windows_x86"
             use_qemu: false
           - os: macos-14
-            arch: "arm64 universal2 x86_64"
+            arch: "universal24"
```
Comment (nightlark) [inline .github/workflows/build.yml:75]: Should this be `universal2`?

Comment (henryiii) [reply .github/workflows/build.yml:75]:
```suggestion
            arch: "universal"
```

Yes 🤦

**Comment by henryiii** [comment]
Do you really want to support manylinux1? It's been abandoned by most projects & is years past EoL. The oldest cmake we still ship is 2010 (and _highly_ recommend 2014). I can require binaries and see what the newest cmake to support manylinux1 was.

**Comment by henryiii** [comment]
Hopefully I remembered how to do that. :)

**Comment by henryiii** [comment]
FYI, the last cmake we shipped for manylinux1 was 3.22.6.

**Comment by nightlark** [comment]
For supporting manylinux1 it's mostly been if the builds work with minimal hassle, might as well keep building them for users that haven't been able to update their Python/OS yet -- not sure if I'm reading it right, but from https://mayeut.github.io/manylinux-timeline/ it looks like there's maybe 3-5% of users with Python <=3.7 that only support manylinux1.

File: pyproject.toml [hunk 2]
```diff
@@ -39,6 +21,28 @@ classifiers = [
...
+wheel.install-dir = "swig/data"
+
+[tool.cibuildwheel]
+build = "cp39-*"
+test-command = [
+  "swig -version",
+  "swig -pcreversion",
+  "swig -help",
+]
+environment.PIP_ONLY_BINARY = ":all"
```
Comment (henryiii) [inline pyproject.toml:39]:
```suggestion
environment.PIP_ONLY_BINARY = ":all:"
```

**Review by nightlark — APPROVED** [review]

**Pull request merged**